File: csharp/Base64Encode.cs
using System.Text;

public class Base64Encode : Benchmark
{
    private int _n;
    private byte[] str1bytes;
    private string _str2 = "";
    private uint _result;

    public Base64Encode()
    {
        _result = 0;
        _n = (int)ConfigVal("size");
    }

    public override void Prepare()
    {
        string _str = new string('a', _n);
        str1bytes = Encoding.UTF8.GetBytes(_str);
        _str2 = Convert.ToBase64String(str1bytes);
    }

    public override void Run(long IterationId) {
        _str2 = Convert.ToBase64String(str1bytes);
        _result += (uint)_str2.Length;
    }

    public override uint Checksum
    {
        get
        {
            string _str = Encoding.UTF8.GetString(str1bytes);
            string resultStr = $"encode {(_str.Length > 4 ? _str.Substring(0, 4) + "..." : _str)} to {(_str2.Length > 4 ? _str2.Substring(0, 4) + "..." : _str2)}: {_result}";
            return Helper.Checksum(resultStr);
        }
    }
}

File: swift/Sources/benchmarks/Base64Encode.swift
import Foundation

final class Base64Encode: BenchmarkProtocol {
    private var sizeVal: Int64 = 0
    private var bytes: [UInt8] = []
    private var str2: String = ""
    private var resultVal: UInt32 = 0

    init() {
        sizeVal = configValue("size") ?? 0
    }

    func prepare() {
        let chars = [Character](repeating: "a", count: Int(sizeVal))
        let str = String(chars)

        bytes = [UInt8](str.data(using: .utf8)!)
        let data = Data(bytes: bytes, count: bytes.count)
        str2 = data.base64EncodedString()
    }

    func run(iterationId: Int) {
        let data = Data(bytes: bytes, count: bytes.count)
        str2 = data.base64EncodedString(options: [])
        resultVal &+= UInt32(str2.count)
    }

    var checksum: UInt32 {
        let str = String(bytes: bytes.prefix(5), encoding: .utf8) ?? "-"
        let truncatedStr = str.count > 4 ? String(str.prefix(4)) + "..." : str
        let truncatedStr2 = str2.count > 4 ? String(str2.prefix(4)) + "..." : str2
        let message = "encode \(truncatedStr) to \(truncatedStr2): \(resultVal)"
        return Helper.checksum(message)
    }
}